lib/xnvme_be_macos_dev.c

```diff
@@ -75,7 +75,7 @@ _get_nvme_smart_interface(io_object_t ioservice_device,
 		(**plugin_interface)
 			->QueryInterface(*plugin_interface,
 					 CFUUIDGetUUIDBytes(kIONVMeSMARTInterfaceID),
-					 nvme_smart_interface);
+					 (void **)*nvme_smart_interface);
```

Comment:
This doesn't look right. It's supposed to be a pointer-pointer. https://developer.apple.com/documentation/iokit/ionvmesmartinterface/3521170-queryinterface

I'm getting segfaults on this locally, while reverting it to `(void **)nvme_smart_interface` is fine.

Reply:
Then that's not the problem. Thanks for testing.

Reply:
No problem :)